### PR TITLE
Closes: #6773 Refactor admin 

### LINKED ISSUE #6773: 3.17 - Refactor Admin
# Description:

This issue focuses on the proposed refactor of the Admin process within the `AboveTheFold` feature. The goal is to make the `Admin` process more generic and reusable for other features. This involves moving the `Admin` process from the `AboveTheFold` namespace to the `PerformanceHints` namespace within `Common`. The `Common` namespace will serve as a central location for processes that are generic and shared across features.

# Scope a solution:

- Move the `Admin` folder from `AboveTheFold` to `PerformanceHints` within `Common`.
- Remove all references to the Admin module in `AboveTheFold\ServiceProvider`
- Update the `Common\PerformanceHints\ServiceProvider` if created and if not create it and reflect the new location of the Admin module.
- Thoroughly test the `Admin` process in its new location to ensure it works as expected and maintains its functionality.
- Update any documentation or comments in the codebase to reflect the change in location of the `Admin` process.

# Acceptance Criteria
- There should be no regression for this module
- All tests should be green and passing.

### PR DESCRIPTION
**Pull request by Khadreal**
# Description
Move service provider for admin to common/performancehints namespace

Fixes #6773 

### User documentation

*Explain how this code impacts users.*

### Technical documentation

*Explain how this code works. Diagram & drawings are welcomed.*

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality).

## New dependencies
*List any new dependencies that are required for this change.*

## Risks
*List possible performance & security issues or risks, and explain how they have been mitigated.*

# Checklists

## Feature validation
- [x] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.

## Documentation
- [ ] I prepared the user documentation for the feature/enhancement and shared it in the PR or the GitHub issue.
- [ ] The user documentation covers new/changed entry points (endpoints, WP hooks, configuration files, ...).
- [ ] I prepared the technical documentation if needed, and shared it in the PR or the GitHub issue.

## Code style
- [ ] I wrote self-explanatory code about what it does.
- [ ] I wrote comments to explain why it does it.
- [ ] I named variables and functions explicitely.
- [ ] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [ ] I listed the introduced external dependencies explicitely on the PR.
- [ ] I validated the repo-specific guidelines from CONTRIBUTING.md.

## Observability
- [ ] I handled errors when needed.
- [ ] I wrote user-facing messages that are understandable and provide actionable feedbacks.
- [ ] I prepared ways to observe the implemented system (logs, data, etc.).

## Risks
- [ ] I explicitely mentioned performance risks in the PR.
- [ ] I explicitely mentioned security risks in the PR.

### CODE REVIEW

File: inc/Engine/Common/PerformanceHints/Admin/ServiceProvider.php
Comment (remyperona) [inline inc/Engine/Common/PerformanceHints/Admin/ServiceProvider.php:1]: that change doesn't make sense, why are we moving this service provider?

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/Engine/Common/PerformanceHints/Admin/ServiceProvider.php
Comment (Khadreal) [reply inc/Engine/Common/PerformanceHints/Admin/ServiceProvider.php:1]: I misunderstood this part `Move the Admin folder from AboveTheFold to PerformanceHints within Common.` of the  grooming, @Miraeld explained this, I'll modify and push the changes

File: inc/Engine/Common/PerformanceHints/Admin/AdminContext.php
Comment (jeawhanlee) [inline inc/Engine/Common/PerformanceHints/Admin/AdminContext.php:1]: We should change this back to `Controller`
Also for the sake of keeping this review short and precise, the contents of the previous Admin\Controller class should not change much, we only need to take out the context interface and the dependency as we would have those in the factory, the method names also needs to change, all occurrence of `atf` should become `performance_hints` and hooks should be updated with backward compatibility using rocket_apply_filter_and_deprecated for filters or rocket_do_action_and_deprecated for actions.

File: inc/Engine/Common/PerformanceHints/Admin/ControllerInterface.php
Comment (jeawhanlee) [inline inc/Engine/Common/PerformanceHints/Admin/ControllerInterface.php:1]: I don't think we need this interface.

File: inc/Engine/Common/PerformanceHints/FactoryInterface.php
Comment (jeawhanlee) [inline inc/Engine/Common/PerformanceHints/FactoryInterface.php:1]: I also don't think we need the changes here.

File: inc/Engine/Media/AboveTheFold/Admin/Controller.php [hunk 1]
```diff
@@ -42,116 +43,13 @@ public function __construct( ATFTable $table, ATFQuery $query, Context $context
...
-		];
-	}
 
 	/**
 	 * Cleans rows for the current URL.
 	 *
 	 * @return void
 	 */
-	public function clean_url() {
+	public function clean_url(): void {
```
Comment (jeawhanlee) [inline inc/Engine/Media/AboveTheFold/Admin/Controller.php]: We shouldn't retain this method in the AboveTheFold Namespace.

File: inc/Engine/Media/AboveTheFold/Admin/Subscriber.php [hunk 2]
```diff
@@ -29,63 +29,10 @@ public function __construct( Controller $controller ) {
...
-			'wp_trash_post'                 => 'delete_post_atf',
-			'delete_post'                   => 'delete_post_atf',
-			'clean_post_cache'              => 'delete_post_atf',
-			'wp_update_comment_count'       => 'delete_post_atf',
-			'edit_term'                     => 'delete_term_atf',
-			'pre_delete_term'               => 'delete_term_atf',
-			'rocket_saas_clean_all'         => 'truncate_atf_admin',
-			'rocket_saas_clean_url'         => 'clean_url',
-			'wp_rocket_upgrade'             => [ 'truncate_on_update', 10, 2 ],
+			'rocket_saas_clean_url' => 'clean_url',
```
Comment (jeawhanlee) [inline inc/Engine/Media/AboveTheFold/Admin/Subscriber.php]: This would also not need to be here anymore.

File: inc/Engine/Media/AboveTheFold/Factory.php
Comment (jeawhanlee) [inline inc/Engine/Media/AboveTheFold/Factory.php:1]: We don't need an admin controller here.

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Review by Miraeld — APPROVED** [review]

**Pull request merged**